### PR TITLE
[INCR-22] Support testing Python 3 in edx-oauth2-provider

### PR DESCRIPTION
**Pull request by jonkiparsky**
[JIRA link](https://openedx.atlassian.net/browse/INCR-22)

Changes:
- update tox.ini per changes by bbearce on his previous attempt at this work (https://github.com/edx/edx-oauth2-provider/pull/56)
- modify target `test` in Makefile, which was attempting to run coverage as well as tests
- bump some versions in test_requirements.txt to allow `make test` and `make coverage` to run
- add some notes in README.txt

Note: `make test-all` runs with failures, as expected.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @jonkiparsky! I've created [OSPR-3450](https://openedx.atlassian.net/browse/OSPR-3450) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@jonkiparsky Thank you for your contribution. Please let me know once all tests are green.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @jonkiparsky! I've created [OSPR-3517](https://openedx.atlassian.net/browse/OSPR-3517) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by jonkiparsky** [comment]
@jmbowman Sorry about the delay on this, but I think this is ready for you now (getting all and only the expected failures)

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = django{18,110,111}
+envlist = py{36,37}-django{111,20,21,22}
```
Comment (jmbowman) [inline tox.ini]: Should probably add py27 to this also.

File: tox.ini [hunk 2]
```diff
@@ -12,9 +12,10 @@ norecursedirs = .*
 
 [testenv]
 deps =
-    django18: Django>=1.8,<1.9
-    django110: Django>=1.10,<1.11
     django111: Django>=1.11,<2.0
+    django20: Django>=2.0,<2.1
+    django21: Django>=2.1,<2.2
+    django22: Django>=2.2
```
Comment (jmbowman) [inline tox.ini]: This should have an upper bound in preparation for when the next major version of Django is released.

File: .travis.yml [hunk 3]
```diff
@@ -2,11 +2,14 @@ language: python
...
+  - 3.6
+  - 3.7
 
 env:
-  - TOXENV=django18
-  - TOXENV=django110
   - TOXENV=django111
+  - TOXENV=django20
+  - TOXENV=django21
+  - TOXENV=django22
```
Comment (jmbowman) [inline .travis.yml]: The failing test workers should be flagged as ok to fail via `allow_failures` until they can be fixed: https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail

**Comment by jonkiparsky** [comment]
@jmbowman Thanks for the feedback. Hope this is satisfactory.

**Comment by natabene** [comment]
@jmbowman This is ready for your review.

**Comment by edx-webhook** [comment]
@jonkiparsky 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-oauth2-provider&entry.867055334=https://github.com/edx/edx-oauth2-provider/pull/58&entry.1484655318=https://github.com/jonkiparsky&entry.752974735=2019-05-11+06:11&entry.1917517419=2019-05-24+14:56&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**